### PR TITLE
[Merged by Bors] - feat(Algebra/Category/AlgebraCat/Monoidal): remove heavy defeqs

### PR DESCRIPTION
**Pull request by yuma-mizuno**
Remove heavy defeqs detected in [zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Heavy.20rfl).

Alternatives for `forget₂_map_associator_hom` and `forget₂_map_associator_inv` are provided in #16778.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by kbuzzard** [comment]
!bench

**Review by jcommelin — COMMENTED** [review]
Negative diff and massive speedup! :octopus: :tada: :rocket: 

bors merge

File: Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean [hunk 1]
```diff
@@ -55,43 +55,14 @@ instance : MonoidalCategoryStruct (AlgebraCat.{u} R) where
...
-        dsimp only [forget₂_module_obj, forget₂_module_map, Iso.refl_symm, Iso.trans_hom,
-          Iso.refl_hom, MonoidalCategory.tensorIso_hom]
-        erw [Category.id_comp, MonoidalCategory.tensor_id, Category.id_comp]
-        rfl
-      rightUnitor_eq := fun X => by
-        dsimp
-        erw [Category.id_comp, MonoidalCategory.tensor_id, Category.id_comp]
-        exact congr_arg LinearEquiv.toLinearMap <|
-          TensorProduct.AlgebraTensorModule.rid_eq_rid R X }
+      associator_eq := fun X Y Z => TensorProduct.ext₃ (fun x y z => rfl)
```
Comment (eric-wieser) [inline Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:63]: I'm rather surprised that this PR "removes heavy defeqs" by using a rfl-based proof!

File: Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean [hunk 2]
```diff
@@ -55,43 +55,14 @@ instance : MonoidalCategoryStruct (AlgebraCat.{u} R) where
...
-        (forget₂ _ (ModuleCat R) |>.obj Z)).hom := by
-  rfl
-
-theorem forget₂_map_associator_inv (X Y Z : AlgebraCat.{u} R) :
-    (forget₂ (AlgebraCat R) (ModuleCat R)).map (α_ X Y Z).inv =
-      (α_
-        (forget₂ _ (ModuleCat R) |>.obj X)
-        (forget₂ _ (ModuleCat R) |>.obj Y)
-        (forget₂ _ (ModuleCat R) |>.obj Z)).inv := by
-  rfl
```
Comment (eric-wieser) [inline Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:72]: Were these lemmas themselves bad? I think when I wrote them, I wanted to characterize how `forget₂ (AlgebraCat R) (ModuleCat R)` commutes with the associator. Is there a way of recovering this elsewhere in the monoidal API, by bundling forget as a monoidal functor?

Comment (yuma-mizuno) [reply Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:72]: The "more type correct" lemmas are those added in the companion PR [#16778](https://github.com/leanprover-community/mathlib4/pull/16778).

Comment (eric-wieser) [reply Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:72]: Thanks! I hadn't expected the complexity that ends up there, but I guess nor am I surprised by it.

Do you know if we have the bundled `forget₂ (AlgebraCat R) (ModuleCat R)` as a `MonoidalFunctor`?

Comment (yuma-mizuno) [reply Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:72]: When transporting the associator, we need to insert `F.μ : F.obj X ⊗ F.obj Y ⟶ F.obj (X ⊗ Y)` appropriate way in general. But for `forget₂` we can ignore it thanks to the defeq property. But it turned out that this defeq was very heavy.

Comment (yuma-mizuno) [reply Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean:72]: > Do you know if we have the bundled forget₂ (AlgebraCat R) (ModuleCat R) as a MonoidalFunctor?

Yes, it's defined at the bottom of this file (`toModuleCatMonoidalFunctor`).

**Pull request closed without merging**